### PR TITLE
use rhel packages for rhel instead of centos ones

### PR DESCRIPTION
**Pull request by sgalsaleh**

### CODE REVIEW

File: bundles/docker-rhel7/Dockerfile [hunk 1]
```diff
@@ -9,6 +8,12 @@ ARG DOCKER_VERSION
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     docker-ce-$(yum list --showduplicates 'docker-ce' | grep ${DOCKER_VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://') \
     docker-ce-cli-$(yum list --showduplicates 'docker-ce-cli' | grep ${DOCKER_VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://')
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:7.4.1708
```
Comment (emosbaugh) [inline bundles/docker-rhel7/Dockerfile:14]: why pin exact version here? how about `centos:7`?

File: bundles/docker-rhel8/Dockerfile [hunk 2]
```diff
@@ -13,6 +11,12 @@ ARG DOCKER_VERSION
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     docker-ce-$(yum list --showduplicates 'docker-ce' | grep ${DOCKER_VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://') \
     docker-ce-cli-$(yum list --showduplicates 'docker-ce-cli' | grep ${DOCKER_VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://')
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:8.1.1911
```
Comment (emosbaugh) [inline bundles/docker-rhel8/Dockerfile:17]: why pin exact version here? how about `centos:8`?

File: bundles/k3s-rhel7/Dockerfile [hunk 3]
```diff
@@ -18,6 +17,12 @@ ARG K3S_VERSION
 RUN envsubst < /tmp/rancher-k3s-common.repo.tmpl > /etc/yum.repos.d/rancher-k3s-common.repo
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     k3s-selinux
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:7.4.1708
```
Comment (emosbaugh) [inline bundles/k3s-rhel7/Dockerfile:23]: why pin exact version here? how about `centos:7`?

File: bundles/k3s-rhel8/Dockerfile [hunk 4]
```diff
@@ -24,6 +20,11 @@ RUN envsubst < /tmp/rancher-k3s-common.repo.tmpl > /etc/yum.repos.d/rancher-k3s-
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     k3s-selinux
 
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:8.1.1911
```
Comment (emosbaugh) [inline bundles/k3s-rhel8/Dockerfile:25]: why pin exact version here? how about `centos:8`?

File: bundles/k8s-rhel7/Dockerfile [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
-RUN yum install -y createrepo
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
 	kubelet-${KUBERNETES_VERSION} \
 	kubectl-${KUBERNETES_VERSION} \
 	kubernetes-cni \
 	git
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:7.4.1708
```
Comment (emosbaugh) [inline bundles/k8s-rhel7/Dockerfile:14]: why pin exact version here? how about `centos:7`?

File: bundles/k8s-rhel8/Dockerfile [hunk 6]
```diff
@@ -1,21 +1,24 @@
...
+RUN yum install -y yum-utils
 RUN yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
 	kubelet-${KUBERNETES_VERSION} \
 	kubectl-${KUBERNETES_VERSION} \
 	kubernetes-cni \
 	git
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:8.1.1911
```
Comment (emosbaugh) [inline bundles/k8s-rhel8/Dockerfile:17]: why pin exact version here? how about `centos:8`?

File: bundles/rke2-rhel7/Dockerfile [hunk 7]
```diff
@@ -20,6 +19,11 @@ RUN source /tmp/env; envsubst < /tmp/rancher-rke2.repo.tmpl > /etc/yum.repos.d/r
 RUN source /tmp/env; yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     rke2-server-${rke2_rpm_version} \
     rke2-agent-${rke2_rpm_version}
-RUN createrepo /packages/archives
 
-CMD cp -r /packages/archives/* /out/
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:7.4.1708
```
Comment (emosbaugh) [inline bundles/rke2-rhel7/Dockerfile:25]: why pin exact version here? how about `centos:7`?

File: bundles/rke2-rhel8/Dockerfile [hunk 8]
```diff
@@ -26,6 +22,12 @@ RUN source /tmp/env; envsubst < /tmp/rancher-rke2.repo.tmpl > /etc/yum.repos.d/r
 RUN source /tmp/env; yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
     rke2-server-${rke2_rpm_version} \
     rke2-agent-${rke2_rpm_version}
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:8.1.1911
```
Comment (emosbaugh) [inline bundles/rke2-rhel8/Dockerfile:28]: why pin exact version here? how about `centos:8`?

File: scripts/Manifest [hunk 9]
```diff
@@ -9,4 +9,27 @@ KURL_UTIL_IMAGE=replicated/kurl-util:alpha
 KURL_BIN_UTILS_FILE=
 # STEP_VERSIONS array is generated by the server and injected at runtime based on supported k8s versions
 STEP_VERSIONS=("" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "1.15.3" "1.16.4" "1.17.13" "1.18.20" "1.19.12", "1.20.8")
-INSTALLER_YAML=
+INSTALLER_YAML=apiVersion: "cluster.kurl.sh/v1beta1
```
Comment (emosbaugh) [inline scripts/Manifest:12]: remove this

File: addons/containerd/template/Dockerfile.rhel7 [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+ARG VERSION
+ENV VERSION=${VERSION}
+
+CMD yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
+    containerd.io-$(yum list --showduplicates 'containerd.io' | grep ${VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://')
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:7.4.1708
```
Comment (emosbaugh) [inline addons/containerd/template/Dockerfile.rhel7:14]: why pin exact version here? how about `centos:7`?

File: addons/containerd/template/Dockerfile.rhel8 [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+ARG VERSION
+ENV VERSION=${VERSION}
+
+CMD yumdownloader --installroot=/tmp/empty-directory --releasever=/ --resolve --destdir=/packages/archives -y \
+    containerd.io-$(yum list --showduplicates 'containerd.io' | grep ${VERSION} | tail -1 | awk '{ print $2 }' | sed 's/.\://')
+
+# installing "createrepo" package on rhel requires a subscription. it's only needed to create the repo,
+# and it's not part of the produced packages, so we can do this:
+FROM centos:8.1.1911
```
Comment (emosbaugh) [inline addons/containerd/template/Dockerfile.rhel8:17]: why pin exact version here? how about `centos:8`?

File: addons/containerd/1.4.4/Manifest [hunk 12]
```diff
@@ -1,6 +1,9 @@
-dockerout rhel-7 addons/containerd/template/Dockerfile.centos7 1.4.4
-dockerout rhel-7-force addons/containerd/template/Dockerfile.centos7-force 1.4.4
-dockerout rhel-8 addons/containerd/template/Dockerfile.centos8 1.4.4
+dockerout centos-7 addons/containerd/template/Dockerfile.centos7 1.4.4
+dockerout centos-7-force addons/containerd/template/Dockerfile.centos7-force 1.4.4
+dockerout centos-8 addons/containerd/template/Dockerfile.centos8 1.4.4
+dockerout rhel-7 addons/containerd/template/Dockerfile.rhel7 1.4.4
+dockerout rhel-7-force addons/containerd/template/Dockerfile.rhel7-force 1.4.4
```
Comment (emosbaugh) [inline addons/containerd/1.4.4/Manifest:5]: i dont think there is any reason to build this bundle and itll just make the packages even bigger. same goes for all instances of rhel-7-force in this pr

File: scripts/common/host-packages.sh [hunk 13]
```diff
@@ -65,7 +65,12 @@ function _rpm_force_install_host_packages() {
     logStep "Installing host packages ${packages[*]}"
 
     local fullpath=
-    fullpath="$(realpath "${dir}")/rhel-7-force${dir_prefix}"
+    if [ "${LSB_DIST}" = "rhel" ]; then
+        fullpath="$(realpath "${dir}")/rhel-7-force${dir_prefix}"
+    else
+        fullpath="$(realpath "${dir}")/centos-7-force${dir_prefix}"
+    fi
```
Comment (emosbaugh) [inline scripts/common/host-packages.sh:72]:
```suggestion
    fullpath="$(realpath "${dir}")/centos-7-force${dir_prefix}"
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/host-packages.sh (hunk 13 above)
Comment (sgalsaleh) [reply scripts/common/host-packages.sh:72]: why?

File: addons/containerd/1.4.4/Manifest (hunk 12 above)
Comment (sgalsaleh) [reply addons/containerd/1.4.4/Manifest:5]: oh we do it for centos because we use that for amazon ol?

File: scripts/common/host-packages.sh [hunk 14]
```diff
@@ -64,8 +64,8 @@ function _rpm_force_install_host_packages() {
 
     logStep "Installing host packages ${packages[*]}"
 
-    local fullpath=
-    fullpath="$(realpath "${dir}")/rhel-7-force${dir_prefix}"
+    local fullpath="$(realpath "${dir}")/centos-7-force${dir_prefix}"
```
Comment (emosbaugh) [inline scripts/common/host-packages.sh:67]: doesnt shellcheck complain about declare and assign in same statement?

**Review by emosbaugh — APPROVED** [review]

**Pull request closed without merging**